projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts

```diff
@@ -0,0 +1,73 @@
+import {TuiDocumentationPagePO, tuiGoto} from '@demo-playwright/utils';
+import {expect, Locator, Page, test} from '@playwright/test';
+
+test.describe(`TablePagination`, () => {
+    test.describe(`Dropdown with \`[size]\`-options (amount items per page)`, () => {
+        test(`Basic case`, async ({page}) => {
+            await tuiGoto(page, `/components/table-pagination/API`);
+            const {apiPageExample} = new TuiDocumentationPagePO(page);
+
+            const {tablePagination, linesPerPageSelect} =
+                await initializeAliases(apiPageExample);
+
+            await linesPerPageSelect.click();
+
+            await waitForCheckmarkIcon(page);
+            await expect(tablePagination).toHaveScreenshot(`0-[size]-dropdown-base.png`);
+        });
+
+        test(`With very long option name`, async ({page}) => {
+            const longNumber = Number.MAX_SAFE_INTEGER;
+
+            await tuiGoto(
+                page,
+                `/components/table-pagination/API?items=[0, ${longNumber}]&size=${longNumber}&total=${longNumber}&page=0`,
+            );
+            const {apiPageExample} = new TuiDocumentationPagePO(page);
+            const {tablePagination, linesPerPageSelect} =
+                await initializeAliases(apiPageExample);
+
+            await linesPerPageSelect.click();
+
+            await waitForCheckmarkIcon(page);
+            await expect(tablePagination).toHaveScreenshot(`1-[size]-dropdown-base.png`);
+        });
+    });
+
+    test(`Custom size-option content`, async ({page}) => {
+        await tuiGoto(page, `/components/table-pagination/API`);
+        const {apiPageExample} = new TuiDocumentationPagePO(page);
+        const {linesPerPageSelect} = await initializeAliases(apiPageExample);
+
+        await linesPerPageSelect.click();
+        await waitForCheckmarkIcon(page);
+
+        const dropdown = page.locator(`tui-dropdown`);
+
+        await expect(dropdown).toHaveScreenshot(`2-[sizeOptionContent]-dropdown.png`);
+    });
+
+    async function initializeAliases(
+        apiPageExample: Locator,
+    ): Promise<{tablePagination: Locator; linesPerPageSelect: Locator}> {
+        const tablePagination = apiPageExample.locator(`tui-table-pagination`);
+        const linesPerPageSelect = apiPageExample.locator(
+            `[automation-id=tui-table-pagination__lines-per-page-wrapper] tui-hosted-dropdown`,
+        );
+
+        await expect(tablePagination).toBeVisible();
+        await expect(linesPerPageSelect).toBeVisible();
+
+        return {tablePagination, linesPerPageSelect};
+    }
+
+    async function waitForCheckmarkIcon(page: Page): Promise<void> {
```

Comment:
Let' create `waitForDropdown` inside `TuiTablePaginationPO` and put these code inside this method.

Reply:
fixed in 1a7f7e6d3d63c38ad73aa828bbd637e8afb0dcb4